### PR TITLE
Feature all products

### PR DESCRIPTION
**Pull request by Robertoabr**
#7

### CODE REVIEW

File: client/components/productCard.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import {setSelectedProduct} from '../store/product'
```
Comment (iamannamai) [inline client/components/productCard.js]: Exports should come straight from `../store`'s index file

File: client/store/product.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	type: SET_ALL_PRODUCTS,
+	data
+});
+
+/**
+ * THUNK CREATORS
+ */
+export const getAllProductsThunk = () => {
+	return async dispatch => {
+		const {data} = await axios.get('/products');
```
Comment (iamannamai) [inline client/store/product.js]: try-catch!

File: client/store/index.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 import thunkMiddleware from 'redux-thunk'
 import {composeWithDevTools} from 'redux-devtools-extension'
 import user from './user'
+import product from './product'
 
-const reducer = combineReducers({user})
+const reducer = combineReducers({user, product})
 const middleware = composeWithDevTools(
   applyMiddleware(thunkMiddleware, createLogger({collapsed: true}))
 )
```
Comment (iamannamai) [inline client/store/index.js]: `export * from product`

File: client/components/allProducts.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React, {Component} from 'react';
+import {getAllProductsThunk} from '../store/product';
```
Comment (iamannamai) [inline client/components/allProducts.js]: Exports should come straight from `../store`'s index file

File: client/routes.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ import {withRouter, Route, Switch} from 'react-router-dom'
 import PropTypes from 'prop-types'
 import {Login, Signup, UserHome} from './components'
```
Comment (iamannamai) [inline client/routes.js:5]: Export `AllProducts` from index here as well, instead of from the specific component file on line 7.

**Comment by Robertoabr** [comment]
this has pull req has been updated at 11:30p EST 4/16 to get it to work with the back end code and to reflect the general comments on the last review

**Review by iamannamai — APPROVED** [review]

**Pull request merged**